osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapProcessor.cs

```diff
@@ -192,24 +192,24 @@ private static void applyOffset(ref float position, float amount)
 
         private static void initialiseHyperDash(IBeatmap beatmap)
         {
-            List<CatchHitObject> objectWithDroplets = new List<CatchHitObject>();
+            List<PalpableCatchHitObject> palpableObjects = new List<PalpableCatchHitObject>();
 
             foreach (var currentObject in beatmap.HitObjects)
             {
                 if (currentObject is Fruit fruitObject)
-                    objectWithDroplets.Add(fruitObject);
+                    palpableObjects.Add(fruitObject);
 
                 if (currentObject is JuiceStream)
                 {
-                    foreach (var currentJuiceElement in currentObject.NestedHitObjects)
+                    foreach (var juice in currentObject.NestedHitObjects)
                     {
-                        if (!(currentJuiceElement is TinyDroplet))
-                            objectWithDroplets.Add((CatchHitObject)currentJuiceElement);
+                        if (juice is PalpableCatchHitObject palpableObject && !(juice is TinyDroplet))
```

Comment:
Not a huge fan of the new pattern match to `PalpableCatchHitObject` added here - juice streams should implicitly only have palpable nested objects and anything else is an invariant violation. I'd probably just straight-cast or make `JuiceStream`/`BananaShower` re-expose `NestedHitObjects` with the more-specific type:

```csharp
public new IReadOnlyList<PalpableCatchHitObject> NestedHitObjects => base.NestedHitObjects.Cast<PalpableCatchHitObject>();
```

Reply:
I'm not sure about this because casting doesn't work for `IReadOnlyList` and I have to make it `IEnumerable<PalpableCatchHitObject> NestedHitObjects => ...`.